crates/tabby/src/services/code.rs

```diff
@@ -113,21 +118,67 @@ impl CodeSearch for CodeSearchImpl {
 
     async fn search_in_language(
         &self,
+        git_url: &str,
         language: &str,
         tokens: &[String],
         limit: usize,
         offset: usize,
     ) -> Result<SearchResponse, CodeSearchError> {
         let language_query = self.schema.language_query(language);
         let body_query = self.schema.body_query(tokens);
+
+        let repos = self
+            .repository_access
+            .list_repositories()
+            .await
+            .unwrap_or_default();
+
+        let git_url = Url::parse(git_url)
+            .map(|mut url| {
+                let _ = url.set_password(None);
+                let _ = url.set_username("");
+                url.to_string()
+            })
+            .unwrap_or_else(|_| git_url.to_string());
```

Comment:
1. move stripping logic as part of `closest_match`.
2. git_urls from `RepositoryAccess` also requires stripping.